### PR TITLE
OTT-1356: add support to update "video position" targeting of line item

### PR DESCRIPTION
**Pull request by ashishshinde-pubm**
update_settings.py would be used to take the user inputs.

**Command to execute update script** 
` python3 -m  tasks.update videoposition`

**Commands to check the unit test code coverage**
```
python3 -m unittest tests.test_update
coverage run -m tests.test_update
coverage report -m
coverage html
google-chrome htmlcov/index.html 
```

Test case coverage report - 

```
Name                                                                                      Stmts   Miss  Cover   Missing
-----------------------------------------------------------------------------------------------------------------------
tasks/update.py                                                                             145      5    97%   16-18, 25
```

### CODE REVIEW

**Review by Pubmatic-Dhruv-Sonone — APPROVED** [review]

File: tasks/update.py [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+        if len(sys.argv) != 2:
+            logger.info("Usage: python -m task.update [VideoPosition|UpdatePrice]")
+            logger.info("Example: python -m task.update VideoPosition")
+            return
+
+        update_task = sys.argv[1]
+        color = Color()
+
+        # Use settings based on the command-line argument
+        if update_task == "VideoPosition":
```
Comment (pm-isha-bharti) [inline tasks/update.py:278]: Can we make this case-insensitive, to prevent potential user errors?

File: tasks/update.py [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+        # Dictionary containing line item names and reasons for skipping (not updating) each line item.
+        skip_line_items = {}
+
+        for line_item in line_items:
+            # do not update line-item if targeting information is missing
+            if 'targeting' not in line_item or not line_item['targeting']:
+                skip_line_items[line_item['name']] = "targeting information is missing"
+                continue
+
+            # do not update line-item if targeting.requestPlatformTargeting (Inventory type) information is missing
```
Comment (pm-isha-bharti) [inline tasks/update.py:190]: As `videoPositionTargeting` comes directly under `targeting` object, do we need do reject LI based on this condition? 
Ref: https://developers.google.com/ad-manager/api/reference/v202311/AdRuleService.VideoPositionTargeting

File: tasks/update.py [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+            if 'requestPlatformTargeting' not in line_item['targeting'] or not line_item['targeting']['requestPlatformTargeting']:
```
Comment (pm-isha-bharti) [inline tasks/update.py:191]: As `videoPositionTargeting` comes directly under `targeting` object, do we need do reject LI based on this condition?

Ref: https://developers.google.com/ad-manager/api/reference/v202311/AdRuleService.VideoPositionTargeting

File: tasks/update.py (hunk 1 above)
Comment (ashishshinde-pubm) [reply tasks/update.py]: done

File: tasks/update.py (hunk 3 above)
Comment (ashishshinde-pubm) [reply tasks/update.py]: as discussed, this update utility should be used only for updating line items created by our tool.
In our tool, while creating line items we always set the value for targeting.requestPlatformTargeting hence we will not add this conditions in code

File: tasks/update.py [hunk 4]
```diff
@@ -265,17 +251,17 @@ def update(self):
...
             return
 
-        update_task = sys.argv[1]
+        update_task = sys.argv[1].lower()
+        logger = logging.getLogger(__name__)
         color = Color()
 
         # Use settings based on the command-line argument
-        if update_task == "VideoPosition":
+        if update_task == "videoposition":
```
[the remark this answers is not in the document]
Comment (ashishshinde-pubm) [reply tasks/update.py:264]: I am using lower() function, hence its already case-insensitive
```
update_task = sys.argv[1].lower()
```

**Review by pm-isha-bharti — APPROVED** [review]

**Pull request merged**